VTUL/data-repo
# start validation script

**Adds rake task to check if files in published datasets were modified after the dataset publish date.**
* * *

**JIRA Ticket**: (https://webapps.es.vt.edu/jira/browse/LIBTD-1819) (:star:)

# What does this Pull Request do? (:star:)
Checks publish dates in csv file provided by Data Curation team to see if files in published datasets were modified after the dataset publish date. Writes results to csv file.

# What's the changes? (:star:)
* Adds `datarepo:validate_published` rake task

# How should this be tested?
* Copy csv file with published dataset info onto data-repo server. CSV file can be found here: https://webapps.es.vt.edu/jira/secure/attachment/188271/20190529_PublishedDatasets.csv
* Create a few collections with titles that match `Dataset Title` and DOI's that match `DOI` fields from  entries in csv file.
* Create several generic files (Items) and attach them to the collections. 
* From the project root, run `./bin/rake datarepo:validate_published["/path/to/your/csv_file"]`
* Check that the output csv file was created in `/var/local/hydra/data-repo/tmp` with the correct values

# Additional Notes:
* branch: `pub_validate`

# Interested parties
@yinlinchen 

(:star:) Required fields
